sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ProductEncoder.scala

```diff
@@ -61,7 +62,31 @@ object ProductEncoder {
 
   def extractorFor(
```

Comment:
BTW, I think that we should undo the split after my PR for REPL classes and reynolds serializer patches are in.